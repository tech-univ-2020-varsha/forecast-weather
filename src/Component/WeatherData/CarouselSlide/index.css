.carousel{
    display: flex;
    flex-direction: column;
    padding: 1em;
    border: solid white;
    margin: 1em;
    justify-content: center;
}

.carouselData{
    align-items: center;
    color: rgba(27, 14, 14, 0.849);
    padding: 0.2em;
}