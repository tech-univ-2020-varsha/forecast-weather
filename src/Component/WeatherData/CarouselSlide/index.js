import React from 'react';
import './index.css';

const CarouselSlide = ({ temp, pressure, humidity }) => (
  <div className="carousel">
    <div className="carouselData">
      T:
      {' '}
      {temp}
    </div>
    <div className="carouselData">
      P:
      {' '}
      {pressure}
    </div>
    <div className="carouselData">
      H:
      {' '}
      {humidity}
    </div>

  </div>
);


export default CarouselSlide;
