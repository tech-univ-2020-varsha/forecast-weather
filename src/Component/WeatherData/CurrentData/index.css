.currentData{
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 1em;
    margin: 1em;
    background-color: rgb(234, 244, 246);
    border:solid gray;
    width: 60%;
    
    height: 10%;
}

.dataItems{
    display: flex;
    padding: 1em;
    margin: 1em;
    border:solid  gray;
    justify-items: center;
    align-items: center;
    text-align: center;
    color: gray;
    font-weight: bold;
}