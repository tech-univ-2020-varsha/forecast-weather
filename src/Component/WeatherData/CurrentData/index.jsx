import React from 'react';
import './index.css';

const CurrentData = ({ temp, pressure, humidity }) => (
  <div className="currentData">
    <div className="dataItems">
      TEMPERATURE:
      {' '}
      {temp}
    </div>
    <div className="dataItems">
      PRESSURE:
      {' '}
      {pressure}
    </div>
    <div className="dataItems">
      HUMIDITY:
      {' '}
      {humidity}
    </div>
  </div>
);


export default CurrentData;
