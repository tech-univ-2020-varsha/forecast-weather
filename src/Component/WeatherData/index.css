.weatherData{
    display: flex;
    height: 20%;
    justify-content: center;
}

.slider{
    margin: 3em;
    padding:2em;
    background-color: rgb(200, 215, 236);

}