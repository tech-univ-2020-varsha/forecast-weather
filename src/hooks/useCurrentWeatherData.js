import { useState, useEffect } from 'react';
import axios from 'axios';

const useCurrentWeatherData = (url) => {
  const [temp, setTemp] = useState(0);
  const [pressure, setPressure] = useState(0);
  const [humidity, setHumidity] = useState(0);
  const [completed, setCompleted] = useState(false);
  useEffect(() => {
    const getData = async () => {
      try {
        const result = await axios.get(url);
        const weatherReport = result.data.main;
        setTemp(weatherReport.temp);
        setPressure(weatherReport.pressure);
        setHumidity(weatherReport.humidity);
        setCompleted(true);
      } catch (err) {
        setCompleted(false);
      }
    };
    getData();
  }, []);
  console.log('temp:', temp, 'press', pressure, 'humidity:', humidity, completed);
  return [temp, pressure, humidity, completed];
};

export default useCurrentWeatherData;
