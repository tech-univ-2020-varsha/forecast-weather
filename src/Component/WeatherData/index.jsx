import React from 'react';
import Slider from 'react-slick';
import useCurrentWeatherData from '../../hooks/useCurrentWeatherData';
import CurrentData from './CurrentData';
import './index.css';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import CarouselSlide from './CarouselSlide';


const WeatherData = () => {
  const latitude = localStorage.getItem('lat') || 0;
  const longitude = localStorage.getItem('long') || 0;
  console.log(latitude, longitude);
  const settings = {
    dots: true,
    infinite: false,
    speed: 500,
    arrows: true,
    slidesToShow: 8,
    slidesToScroll: 1,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 3,
          infinite: true,
          dots: true,
        },
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 2,
          initialSlide: 2,
        },
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        },
      },
    ],
  };
  const apiURL = `http://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=fad7fd3a00475da64538a80de84be99d`;

  const [temp, pressure, humidity, completed] = useCurrentWeatherData(apiURL);
  console.log('completed', completed);
  if (completed) {
    const weatherDataList = JSON.parse(localStorage.getItem('weatherData') || '[]');

    weatherDataList.push({ temp, pressure, humidity });
    // localStorage.setItem('weatherData', JSON.stringify([]));
    localStorage.setItem('weatherData', JSON.stringify(weatherDataList));
    return (
      <div>
        <div className="weatherData">
          <CurrentData temp={temp} pressure={pressure} humidity={humidity} />
        </div>
        <div className="slider">
          <Slider {...settings}>
            { weatherDataList.map((data) => (
              <CarouselSlide temp={data.temp} pressure={data.pressure} humidity={data.humidity} />

            ))}
          </Slider>
        </div>

      </div>

    );
  }

  { return <div>Loading ...</div>; }
};


export default WeatherData;
