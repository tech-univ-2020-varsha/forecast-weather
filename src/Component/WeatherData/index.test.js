import { render } from '@testing-library/react';
import React from 'react';
import WeatherData from './index';
import useCurrentWeatherData from '../../hooks/useCurrentWeatherData';


describe('the getWeather data fruntion', () => {
  useCurrentWeatherData = jest.fn(() => {});
  it('should render correctly', () => {
    const { asFragment } = render(<WeatherData latitude={99.86} longitude={98.5}>  </WeatherData>);
    expect(asFragment).toMatchSnapshot();
  });
});
