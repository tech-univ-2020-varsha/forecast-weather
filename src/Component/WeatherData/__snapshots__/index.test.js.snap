// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the getWeather data fruntion should render correctly 1`] = `[Function]`;
